projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -69,10 +69,11 @@ export abstract class NgxSubFormComponent<ControlInterface, FormInterface = Cont
   // when developing the lib it's a good idea to set the formGroup type
   // to current + `| undefined` to catch a bunch of possible issues
   // see @note form-group-undefined
-  public formGroup: TypedFormGroup<FormInterface> = (new FormGroup(
-    this._getFormControls(),
-    this.getFormGroupControlOptions() as AbstractControlOptions,
-  ) as unknown) as TypedFormGroup<FormInterface>;
+  // formGroup will only be defined after ngOnInit
+  // otherwise it's impossible to access class properties
+  // from the `getFormControls` method which is an issue with validators
+  // see https://github.com/cloudnc/ngx-sub-form/issues/82
+  public formGroup!: TypedFormGroup<FormInterface>;
```

Comment:
I've added `!` as I don't think our users should have to deal with the fact that the formGroup can be undefined.

The cases where it might be an issue would be from the TS part of their components and for ex trying to access `formGroup` in the constructor but it feels like that's not a common use case :thinking: 

Not sure on that one, let me know what you think

Reply:
Discussed IRL, I'm convinced :)

Reply:
Just to clarify my point we'd also need to have an ngIf check in the view with AoT (maybe not in v8 but v9) and if anyone tries to use the formGroup in the constructor they'll get an error straight away so I think this should be pretty safe to do rather than setting up an empty form where people wouldn't get the value they expect out of that form at that point of time (before ngOnInit).